scene/2d/navigation_region_2d.cpp

```diff
@@ -307,6 +307,11 @@ void NavigationRegion2D::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "navigation_layers", PROPERTY_HINT_LAYERS_2D_NAVIGATION), "set_navigation_layers", "get_navigation_layers");
 	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "enter_cost"), "set_enter_cost", "get_enter_cost");
 	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "travel_cost"), "set_travel_cost", "get_travel_cost");
+
+#ifndef DISABLE_DEPRECATED
+	// Compatibility properties
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "navpoly", PROPERTY_HINT_RESOURCE_TYPE, "NavigationPolygon", PROPERTY_USAGE_INTERNAL), "set_navigation_polygon", "get_navigation_polygon");
+#endif // DISABLE_DEPRECATED
```

Comment:
I think usually this would be handled in `_set` and `_get` virtual methods, instead of registering a property like this.